packages/nimble-components/src/rich-text-editor/index.ts

```diff
@@ -11,7 +16,193 @@ declare global {
 /**
  * A nimble styled rich text editor
  */
-export class RichTextEditor extends FoundationElement {}
+export class RichTextEditor extends FoundationElement {
+    /**
+     * @internal
+     */
+    @observable
+    public bold!: ToggleButton;
+
+    /**
+     * @internal
+     */
+    @observable
+    public italics!: ToggleButton;
+
+    /**
+     * @internal
+     */
+    @observable
+    public bulletList!: ToggleButton;
+
+    /**
+     * @internal
+     */
+    @observable
+    public numberedList!: ToggleButton;
+
+    /**
+     * @internal
+     */
+    public editor!: HTMLDivElement;
+
+    private tiptapEditor!: Editor;
+
+    /**
+     * @internal
+     */
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.initializeEditor();
+        this.bindEditorTransactionEvent();
+    }
+
+    /**
+     * @internal
+     */
+    public override disconnectedCallback(): void {
+        super.disconnectedCallback();
+        this.tiptapEditor.off('transaction');
+    }
+
+    /**
+     * Toggle the bold mark and focus back to the editor
+     * @internal
+     */
+    public boldButtonClickHandler(): void {
+        this.tiptapEditor.chain().focus().toggleBold().run();
+    }
+
+    /**
+     * Toggle the bold mark and focus back to the editor
+     * @internal
+     */
+    public boldButtonKeyDownHandler(e: KeyboardEvent): boolean {
+        switch (e.key) {
+            case keySpace:
+            case keyEnter:
+                this.tiptapEditor.chain().focus().toggleBold().run();
+                return false;
+            default:
+                return true;
+        }
+    }
```

Comment:
This function could be named `keyDownActivateHandler` as it always takes space and enter keys to activate any button and kept in the abstract class. Consider implementing this function in the parent class `EditorButton`

In `EditorButton` parent class let's have the common code except for the line 
`this.tiptapEditor.chain().focus().toggleBold().run()`. This particular line can be passed from child classes like `BoldButton`, `ItalicsButton`, etc. as a callback function.

Reply:
I have updated the method name as suggested for the key-down handler. I was able to break the `chain()` and moved the focus to the base class.